docs/BOOTING.md

```diff
@@ -26,6 +26,29 @@ sections walk you through various boot sequences all using a single EVE disk ima
 Make sure to familiarize yourself with the implementation details of [make-raw](../pkg/mkimage-raw-efi/make-raw)
 since it is the main entry point into how EVE formats the disks it expects to be booted from.
 
+## Boot time artifacts
+
+Before we proceed describing each type of boot flow in details, let us first establish what kind of boot artifacts are required for EVE to boot. These are:
+
+1. rootfs image (either a [squashfs](https://www.kernel.org/doc/html/latest/filesystems/squashfs.html) or [ext4](https://www.kernel.org/doc/html/latest/filesystems/ext4/index.html))
+2. 2nd stage GRUB (in UEFI format)
+3. type-1 hypervisor ([Xen](https://xenproject.org/) or [ACRN](https://projectacrn.org/))
+4. Dom0 linux kernel (this doubles as type-2 hypervisor for KVM)
+5. Optional Alpine-derived initramfs [init entrypoint](https://gitlab.alpinelinux.org/alpine/mkinitfs) as a compressed cpio
+6. Optional container that performs [EVE installation](../pkg/mkimage-raw-efi/install) as a compressed cpio
+7. Optional extra files (e.g. content of config partition) as a compressed cpio
+
+While EVE bootable images come in at least two different flavors [installer](https://github.com/lf-edge/eve/blob/master/docs/BUILD.md#using-an-installer-image) and [live](https://github.com/lf-edge/eve/blob/master/docs/BUILD.md#installing-a-live-image) the boot flow and boot artifacts are all the same. Therefore, the only difference between an installer and a live image are all the optional boot artifacts #5-#7. In fact, the distinction is so blurry that an installer image that booted on the box and put all the required bits on disk can continue running effectively becoming a live image without the need for a reboot.
+
+Since artifacts #2-#5 are physically embedded inside of the rootfs image (artifact #1) a capable firmware (such as latest versions of u-boot) should be able to boot EVE with only having access to that single binary. However, since we have to deal with less capable firmware implementations (such as UEFI and legacy BIOS - see below) we have to leverage 1st stage GRUB or iPXE to parse rootfs image it in order to get access to artifacts #2-#5. This parsing can happen either via reading required portions of rootfs image from disk on demand (via GRUB's [loopback booting](https://www.gnu.org/software/grub/manual/grub/html_node/Loopback-booting.html)) or by first loading rootfs image to memory (via iPXE extensions). The fact that the size of rootfs image is deliberately kept very small, makes both of these methods roughly similar when it comes to performance and overhead. Still, it is an overhead and whenever possible EVE tries to rely on rootfs image sitting on a local disk so that it doesn't have to waste precious memory for hosting its. This, of course, creates a bit of a complication where one following the bootflow has to constantly be aware whether rootfs is on disk or in memory.
```

Comment:
Second to last sentence ends either with a type or missing some word(s):

> precious memory for hosting its.

I am not sure what that is supposed to be.

Reply:
Artifacts 2-5 are always embedded in artifact 1? I think it might make sense to show them as sub-list-items, indented, sort of:

1. rootfs image (either a [squashfs](https://www.kernel.org/doc/html/latest/filesystems/squashfs.html) or [ext4](https://www.kernel.org/doc/html/latest/filesystems/ext4/index.html)), which contains the following
   1. 2nd stage GRUB (in UEFI format)
   1. type-1 hypervisor ([Xen](https://xenproject.org/) or [ACRN](https://projectacrn.org/))
   1. Dom0 linux kernel (this doubles as type-2 hypervisor for KVM)
   1. Optional Alpine-derived initramfs [init entrypoint](https://gitlab.alpinelinux.org/alpine/mkinitfs) as a compressed cpio
1. Optional container that performs [EVE installation](../pkg/mkimage-raw-efi/install) as a compressed cpio
1. Optional extra files (e.g. content of config partition) as a compressed cpio

Reply:
Some questions about the rootfs:

* If the rootfs is physically embedded inside the rootfs, I don't get how the Alpine-derived initramfs is optional? Isn't it always there or always not? We have a single `make rootfs` command
* Is it correct that there is one other part of the rootfs, the actual filesystem itself?

Reply:
> 6. Optional container that performs [EVE installation](../pkg/mkimage-raw-efi/install) as a compressed cpio

Can we describe how this works, what creates it, where?

Reply:
> Second to last sentence ends either with a type or missing some word(s):

Fixed.

> Artifacts 2-5 are always embedded in artifact 1?

Most of the time -- hence I don't want to turn it into a physical diagram just yet

> Some questions about the rootfs

The way to think about it is this: all required boot artifacts are inside of rootfs, BUT sometimes we also stick them in other places to interface with silly firmwares.

> Can we describe how this works, what creates it, where?

For sure -- but as we've agreed -- we're changing it right after this PR gets merged -- so I don't want to document what is going to change ASAP.